packages/frontend/src/components/AddAppConnection/index.tsx

```diff
@@ -41,12 +42,18 @@ export default function AddAppConnection(
     : auth?.authenticationSteps
 
   React.useEffect(() => {
-    if (window.opener) {
-      window.opener.postMessage({
-        source: 'plumber',
-        payload: window.location.search,
-      })
-      window.close()
+    if (
+      // checks if this window/popup is opened by the same origin
+      getOpenerOrigin() === window.location.origin
```

Comment:
UPDATE: actually wait...... i think menlo can break this. Cos their URL is always prefixed with the menlo domain